pkg/authorization/api/validation/validation.go

```diff
@@ -8,48 +8,126 @@ import (
 	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
 )
 
-// ValidatePolicyBinding tests required fields for a PolicyBinding.
-func ValidatePolicyBinding(policyBinding *authorizationapi.PolicyBinding) fielderrors.ValidationErrorList {
+func ValidatePolicyName(name string, prefix bool) (bool, string) {
+	if name != authorizationapi.PolicyName {
+		return false, "name must be " + authorizationapi.PolicyName
+	}
+
+	return true, ""
+}
+
+func ValidatePolicy(policy *authorizationapi.Policy) fielderrors.ValidationErrorList {
 	allErrs := fielderrors.ValidationErrorList{}
+	allErrs = append(allErrs, validation.ValidateObjectMeta(&policy.ObjectMeta, true, ValidatePolicyName).Prefix("metadata")...)
+
+	for roleKey, role := range policy.Roles {
+		if roleKey != role.Name {
+			allErrs = append(allErrs, fielderrors.NewFieldInvalid("roles."+roleKey+".name", policy, "must be "+roleKey))
```

Comment:
Not necessarily.  The slices have names associated with them as well.  I hadn't thought of it before, but that doesn't make a lot of sense and should probably be changed in v1beta2.  However, for now, it is possible to assign a role to key "foo" with name "bar" and that won't be caught by conversion.